_im-plugin/notifications-settings.md

```diff
@@ -0,0 +1,163 @@
+---
+layout: default
+title: Configuring notifications settings
+parent: Managing indexes
+nav_order: 100
+---
+
+## Configuring notifications settings
+
+Introduced 2.8
+{: .label .label-purple }
+
+Notifications settings allow users to control how they receive notifications about index events. You can use either OpenSearch Dashboards or the REST API to configure notifications. Supported communication channels include Amazon Chime, Amazon Simple Notification Service (Amazon SNS), Amazon Simple Email Service (Amazon SES), email through SMTP, Slack, and custom webhooks. Supported index operations include resize, reindex, open, and force merge. See [Notifications](https://opensearch.org/docs/latest/observing-your-data/notifications/index/) for further details. 
+
+To configure configure the notifications settings for long-running index operations using OpenSearch Dashboards, go to **Dev Tools** under the main menu. When creating notifications using the API, the LRON setting can be configured two ways: as a concrete task or as a global and persistent task. The LRON setting with `task_id` is ad hoc and automatically deletes when the task ends. The LRON setting using `action_name` is global and persists and applies to all operations of this action type.
```

Comment:
I think the reference to Dev Tools in Dashboards is confusing. The way to do it using the UI is to go to index management > Notifications. If the user is doing it through the API, they can use curl or dev tools. Normally, we don't direct them to dev tools from an individual API page. How about
```suggestion
Configuring notifications settings is useful for long-running index operations. When creating long-running operation notifications using the API, you can configure the `lron_config` setting in two ways:
- As a one-time task: If you pass a `task_id` in the `lron_config` object, the task is one-time and the setting automatically deletes when the task ends.
- As a global and persistent task: If you pass an `action_name` in the `lron_config` object, the task is global, persistent, and applies to all operations of this action type.
```

Reply:
@zhichao-aws Some questions:
- Where do you get the `task_id` from?
- In the example below, both `task_id` and `action_name` are passed. What happens in this case? 
- For the request and response, we have to provide descriptions for all fields and the possible values for enum fields. Could you help with that? Also, what are the possible values for action_name? For `lron_condition`, are the possible values only `success` and `failure`? Which fields are required and which are optional?

Reply:
- The task id can be obtained by setting the request parameter `wait_for_completion` as `false` for open/reindex/resize/force merge. https://github.com/opensearch-project/OpenSearch/issues/6228. The index operation will return the task id immediately and execute at the background.
- Infact if task_id has higher priority. If task_id is passed, action_name will be ignored in our code. But it may be useful if user want to search settings about certain actions.
- For lron_condition, it only has `success` and `failure`. Both of them are optional, the value is `true` by default. `lron_condition` itself is optional (true, true by default). For action_name, current possible values are "indices:data/write/reindex", "indices:admin/resize", "indices:admin/forcemerge", "indices:admin/open".

Reply:
The config must contain taskID or actionName. And enabled lron_config(i.e. at least there is one `true` in `lron_condition`) must contain at least one channel

Reply:
Thanks a lot, @zhichao-aws! Could you also provide an actual example of a request and response for each section where we have them? Thanks!

Reply:
Do you mean create/update/read/delete section, or the cases we talked above?

Reply:
create/update/read/delete. Basically, wherever we have requests/response placeholders now we should have actual requests/responses.

Reply:
create/update are actual requests/responses. Here are actual responses of get/delete:
request:
`GET _plugins/_im/lron`
response:
```
{
  "lron_configs": [
    {
      "_id": "LRON:indices:admin/open",
      "lron_config": {
        "lron_condition": {
          "success": false,
          "failure": false
        },
        "action_name": "indices:admin/open",
        "channels": []
      }
    },
    {
      "_id": "LRON:indices:data/write/reindex",
      "lron_config": {
        "lron_condition": {
          "success": false,
          "failure": true
        },
        "action_name": "indices:data/write/reindex",
        "channels": [
          {
            "id": "my_chime"
          }
        ]
      }
    }
  ],
  "total_number": 2
}
```
request:
`GET _plugins/_im/lron/LRON:indices:data%2Fwrite%2Freindex`
response:
```
{
  "lron_configs": [
    {
      "_id": "LRON:indices:data/write/reindex",
      "lron_config": {
        "lron_condition": {
          "success": false,
          "failure": true
        },
        "action_name": "indices:data/write/reindex",
        "channels": [
          {
            "id": "my_chime"
          }
        ]
      }
    }
  ],
  "total_number": 1
}
```
request:
`DELETE _plugins/_im/lron/LRON:indices:data%2Fwrite%2Freindex`
response(same with delete doc in opensearch normal index):
```
{
  "_index": ".opensearch-control-center",
  "_id": "LRON:indices:data/write/reindex",
  "_version": 2,
  "result": "deleted",
  "forced_refresh": true,
  "_shards": {
    "total": 2,
    "successful": 2,
    "failed": 0
  },
  "_seq_no": 2,
  "_primary_term": 1
}
```